### PR TITLE
convert rgb image to grayscale

### PR DESCRIPTION
**Pull request by BigRedT**
simple function to convert rgb image in ndarray to grayscale

### CODE REVIEW

File: aiutils/vis/image_io.py [hunk 1]
```diff
@@ -40,3 +40,39 @@ def imshow(np_im):
...
+
+def imwrite(np_im, filename):
+    """
+    Matlab like function for displaying a numpy ndarray as an image
+
+    Args:
+    np_im (numpy.ndarray): h x w x 3 ndarray for color images and 
+        h x w for grayscale images with pixels stored in uint8 format
+    """
+    err_str = 'imshow expects ndarray of dimension h x w x c (RGB) or h x w (L)'
```
Comment (jrock08) [inline aiutils/vis/image_io.py]: imwrite

File: aiutils/vis/image_io.py [hunk 2]
```diff
@@ -40,3 +40,39 @@ def imshow(np_im):
...
+    assert (len(np_im.shape) == 3 or len(np_im.shape) == 2), err_str
+
+    if len(np_im.shape) == 3:
+        assert (np_im.shape[2] == 3), 'imshow expected 3 channels'
```
Comment (jrock08) [inline aiutils/vis/image_io.py]: ditto

File: aiutils/vis/image_io.py [hunk 3]
```diff
@@ -40,3 +40,39 @@ def imshow(np_im):
...
+
+
+def rgb2gray(np_im):
+    """
+    Matlab like function for converting numpy ndarray from RGB to grayscale
+
+    Args:
+    np_im (numpy.ndarray): h x w x 3 ndarray storing a color image
+    """
+    err_str = 'imshow expects ndarray of dimension h x w x 3 (RGB)'
```
Comment (jrock08) [inline aiutils/vis/image_io.py]: rgb2gray

**Comment by jrock08** [comment]
Might want to call the module something other than image_io?

File: aiutils/vis/image_io.py [hunk 4]
```diff
@@ -40,3 +41,84 @@ def imshow(np_im):
...
+    assert (len(np_im.shape) == 3), err_str
+
+    im = Image.fromarray(np_im, 'RGB')
+    im_array = np.array(im.convert('L'))
+
+    return im_array
+
+
+def imwrite(np_im, filename):
```
Comment (daeyun) [inline aiutils/vis/image_io.py]: `scipy.misc.imsave`?

http://docs.scipy.org/doc/scipy/reference/generated/scipy.misc.imsave.html
https://github.com/scipy/scipy/blob/fb90c1b7eb9ce0a50453bcba142252662b25e6bb/scipy/misc/pilutil.py#L158
This also uses PIL

**Comment by daeyun** [comment]
Consider using something like OpenCV or [scikit-image](http://scikit-image.org/docs/dev/api/skimage.color.html) for this? OpenCV is a big dependency, but there's a package in Anaconda.

**Comment by jrock08** [comment]
@daeyun PIL isn't a new dependency here, perhaps best to open an issue saying to remove PIL in favor of something else and deal with it outside this commit?

File: aiutils/vis/image_io.py (hunk 4 above)
Comment (BigRedT) [inline aiutils/vis/image_io.py]: @daeyun So the above function can be modified to work for other modes as well. I just covered the most common cases and was hoping people will add them as and when needed -
http://effbot.org/imagingbook/concepts.htm#mode

**Comment by daeyun** [comment]
I think PIL is fine. I was just considering the case of having to add more functions like this, but I agree that can be addressed outside of this.

File: aiutils/vis/image_io.py (hunk 4 above)
Comment (daeyun) [inline aiutils/vis/image_io.py]: sgtm

**Comment by BigRedT** [comment]
@jrock08 @daeyun scikit-image does have a better documentation and more functionality. The functions in this file are some of the first helpers that I wrote when I started using Tensorflow. So if you guys are up to rewriting them using scikit-image, I don't have a problem with it.

**Comment by BigRedT** [comment]
@jrock08 what name do you recommend for the module?

**Comment by kevjshih** [comment]
Would it be different from using the tensorflow version? https://www.tensorflow.org/versions/r0.9/api_docs/python/image.html#rgb_to_grayscale

**Comment by BigRedT** [comment]
@kevjshih wouldn't this require running the op in a session (including feeding an image). Using tensorflow for this might be in an extreme overkill. Also somebody might use these functions for non-tensorflow purposes.

**Comment by kevjshih** [comment]
I suppose that's true.

**Comment by jrock08** [comment]
I'd break this stuff into

image_io.{imread,imwrite,imshow}

image.{rgb2gray... etc}

File: aiutils/vis/image_io.py [hunk 5]
```diff
@@ -1,5 +1,6 @@
 from PIL import Image
 import numpy as np
+import math
```
Comment (jrock08) [inline aiutils/vis/image_io.py]: unused here?

File: aiutils/vis/image.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
+from PIL import Image
+import numpy as np
+import math
+import pdb
```
Comment (jrock08) [inline aiutils/vis/image.py]: unused here

File: aiutils/vis/image_io.py [hunk 7]
```diff
@@ -1,5 +1,6 @@
...
 import pdb
```
Comment (jrock08) [inline aiutils/vis/image_io.py]: unused?

**Pull request merged**